lib/src/source/git.dart

```diff
@@ -306,10 +306,13 @@ class BoundGitSource extends CachedSource {
       var revisionCachePath = _revisionCachePath(id);
       await _revisionCacheClones.putIfAbsent(revisionCachePath, () async {
         if (!entryExists(revisionCachePath)) {
-          await _clone(_repoCachePath(ref), revisionCachePath);
+          await _clone(ref.description['url'], revisionCachePath,
+              reference: _repoCachePath(ref));
           await _checkOut(revisionCachePath, id.description['resolved-ref']);
+          await _pullLFS(revisionCachePath);
           _writePackageList(revisionCachePath, [id.description['path']]);
```

Comment:
Should we consider deleting the folder if one of these steps fails?

Before it wasn't really a problem, because `_clone` and `_checkOut` didn't do any network I/O, so odds that one of these commands was going to fail was probably very small.

But if `_pullLFS` fails, then what? How does the user recover? won't it leave the directory in place with a partially cloned LFS objects..

Reply:
I don't think a folder deletion would be desirable because of the potential size of LFS objects. A likely scenario is:
1. `git clone` succeeds and `revisionCachePath` is created with the lightweight contents of the repo
2. `git lfs pull` starts on the heavy downloads and pulls down 1Gb of a 1.1Gb binary object
3. The network breaks and an error is output to console
4. The developer re-runs `pub get`
5. The `if` checks the existence of the cloned repo and drops to the `else` clause
6. Downloading resumes